source/Calamari/Kubernetes/ResourceStatus/ResourceStatusChecker.cs

```diff
@@ -42,78 +40,64 @@ public ResourceStatusChecker(IResourceRetriever resourceRetriever, IResourceUpda
         }
         
         public bool CheckStatusUntilCompletionOrTimeout(IEnumerable<ResourceIdentifier> resourceIdentifiers, 
-            IStabilizingTimer stabilizingTimer,
-            Kubectl kubectl)
+            ITimer timer,
+            Kubectl kubectl,
+            Options options)
         {
-            var definedResources = resourceIdentifiers.ToList();
-
             var resourceStatuses = new Dictionary<string, Resource>();
             var deploymentStatus = DeploymentStatus.InProgress;
-            var shouldContinue = true;
             var checkCount = 0;
+            var definedResources = resourceIdentifiers.ToList();
             
-            stabilizingTimer.Start();
+            timer.Start();
             
-            while (shouldContinue)
+            do
             {
-                var newResourceStatuses = resourceRetriever
-                    .GetAllOwnedResources(definedResources, kubectl)
+                var newDefinedResourceStatuses = resourceRetriever
+                    .GetAllOwnedResources(definedResources, kubectl, options)
+                    .ToList();
+                var newResourceStatuses = newDefinedResourceStatuses
                     .SelectMany(IterateResourceTree)
                     .ToDictionary(resource => resource.Uid, resource => resource);
 
-                var newDeploymentStatus = GetDeploymentStatus(newResourceStatuses.Values.ToList());
+                var newDeploymentStatus = GetDeploymentStatus(newDefinedResourceStatuses, definedResources);
 
                 reporter.ReportUpdatedResources(resourceStatuses, newResourceStatuses, ++checkCount);
-                
-                shouldContinue = stabilizingTimer.ShouldContinue(deploymentStatus, newDeploymentStatus);
 
                 resourceStatuses = newResourceStatuses;
                 deploymentStatus = newDeploymentStatus;
                 
-                Thread.Sleep(PollingIntervalSeconds * 1000);
-            }
-
-            if (stabilizingTimer.IsStabilizing())
-            {
-                switch (deploymentStatus)
-                {
-                    case DeploymentStatus.Succeeded:
-                        log.Verbose("Resource status check terminated during stabilization period with all resources being successful");
-                        break;
-                    case DeploymentStatus.Failed:
-                        log.Verbose("Resource status check terminated during stabilization period with some failed resources");
-                        break;
-                    default:
-                        break;
-                }
-
-                return false;
+                timer.Tick();
```

Comment:
It wasn't until reading the timer class that I realised `Tick()` is actually a `Sleep`. Consider using the name `Sleep` or `Wait` or something that indicates what its doing. The clock semantics is less important than being clear about what line of code is blocking.